src/coreclr/jit/emitarm64.cpp

```diff
@@ -15596,11 +16892,28 @@ void emitter::emitDispSveRegList(regNumber firstReg, unsigned listSize, insOpts
     regNumber currReg = firstReg;
 
     printf("{ ");
-    for (unsigned i = 0; i < listSize; i++)
+    if (listSize > 1)
     {
-        const bool notLastRegister = (i != listSize - 1);
-        emitDispSveReg(currReg, opt, notLastRegister);
-        currReg = (currReg == REG_V31) ? REG_V0 : REG_NEXT(currReg);
+        if ((listSize == 2) || (((unsigned)currReg + listSize - 1) > (unsigned)REG_V31))
```

Comment:
> (listSize == 2)

Can't this be true for any `listSize > 1`?

Reply:
can you add a unit test that passes `REG_V31` and see if they are disassembled correctly?

Reply:
Sure

Reply:
I already have unit tests that use REG_V31. It prints out `ld4b    { z31.b, z0.b, z1.b, z2.b }, p2/z, [x1, #-0x20, mul vl]`

Reply:
> > (listSize == 2)
> 
> Can't this be true for any `listSize > 1`?

looks like this is not yet addressed.

Reply:
We do need the check, but I slightly changed it.